drivers/flash/spi_nor.c

```diff
@@ -186,6 +186,13 @@ static const struct jesd216_erase_type minimal_erase_types[JESD216_NUM_ERASE_TYP
 };
 #endif /* CONFIG_SPI_NOR_SFDP_MINIMAL */
 
+/* Different operations should poll the WIP bit at different rates */
+enum wait_ready_op_type {
+	WAIT_READY_REGISTER     = 0,
+	WAIT_READY_WRITE        = 1,
+	WAIT_READY_ERASE        = 2,
+};
```

Comment:
I was already thinking on how to address the issue you are addressing here, with addition to hardware timeouts and some other stuff, but there are currently some Zephyr design hurdles I can not overcome easily, and I will say no here.
Just parametrize the `spi_nor_wait_until_ready` to take the timeout directly.
You can still use the above but define them directly as timeouts, i.e. `#define WAIT_READY_REGISTER K_NO_WAIT`.

Reply:
Fair enough, the enum would allow for different backoff schemes, but thats not a high priority